src/Messaging/src/RabbitMQ/Core/RabbitTemplate.cs

```diff
@@ -2266,7 +2266,7 @@ private void LogReceived(IMessage message)
 
         TReply reply = receiveAndReplyCallback(messageAsTReceive);
 
-        if (reply != null)
+        if (reply is not null)
```

Comment:
@hananiel `reply` is of unconstrained generic type `TReply`, so comparing to null doesn't make sense when it's a value type (resulting in S2955 warning). How do you think this should be solved? I've tried adding a class constraint, but that results in errors.

Reply:
@dtillman